arc/job/adapters/torch_ani_test.py

```
@@ -0,0 +1,228 @@
+        self.assertAlmostEqual(calculate_angle(coords=self.job_3.opt_xyz['coords'], atoms=[1, 2, 3], index=1), 109, places=2)
+        self.assertAlmostEqual(calculate_dihedral_angle(coords=self.job_3.opt_xyz['coords'], torsion=[3, 2, 1, 5], index=1),
+                               179.8, places=2)
+
+        self.assertIsNone(self.job_4.opt_xyz)
+        self.job_4.execute()
+        self.assertEqual(self.job_4.opt_xyz['symbols'], ('C', 'C', 'O', 'H', 'H', 'H', 'H', 'H', 'H'))
+
+    def test_run_vibrational_analysis(self):
+        """Test the run_vibrational_analysis() method."""
+        self.job_10.execute()
+        results = read_yaml_file(path=self.job_10.local_path_to_output_file)
+        self.assertEqual(list(results.keys()), ['force_constants', 'freqs', 'hessian', 'modes', 'reduced_masses'])
+        self.assertGreater(len(results['force_constants']), 0)
+        self.assertGreater(len(results['freqs']), 0)
+        self.assertGreater(len(results['hessian'][0][0]), 0)
+        self.assertGreater(len(results['modes'][0][0]), 0)
+        self.assertGreater(len(results['reduced_masses']), 0)
+
+    def test_run_freq(self):
+        """Test the run_freq() method."""
+        self.assertIsNone(self.job_6.freqs)
+        self.assertIsNone(self.job_7.freqs)
+        self.job_6.execute()
+        self.job_7.execute()
+        self.assertAlmostEqual(self.job_6.freqs[-1], 3756.57643, places=3)
+        self.assertGreater(self.job_7.freqs[-1], 3800)
+
+    def test_check_settings(self):
+        """Test the test_check_settings method"""
+        dummy = self.job_1
+        self.assertFalse(dummy.check_settings(settings={"model":"Kinetic Filter and Imputation Regression"}))
+        self.assertFalse(dummy.check_settings(settings={"device":"gpu"}))
+        self.assertFalse(dummy.check_settings(settings={"engine":"Knowledge-based Approach for Linear And Nonlinear optimization"}))
+        self.assertFalse(dummy.check_settings(settings={"fmax":"0.001"}))
```

why is this False?
Can you also test cases where the expected result is True?

---

Nope, they are defined as float or int. I can make a comment about that in the settings, and maybe raise an error when it happens.

---

OK. Can you add tests with correct values where you assert that the `check_settings` method returns `True`?